src/Core/Asset/AnimationData.hpp

```diff
@@ -4,69 +4,102 @@
+#include <Core/Animation/KeyFramedValue.hpp>
+#include <Core/Asset/AnimationTime.hpp>
 #include <Core/Asset/AssetData.hpp>
 #include <Core/RaCore.hpp>
 
 namespace Ra {
 namespace Core {
 namespace Asset {
 
-/// A HandleAnimation stores the animation KeyFrames of transforms applied to
-/// a part of an object, e.g. a bone of an animation skeleton.
+/**
+ * A HandleAnimation stores data for an animation Handle.
+ */
 struct RA_CORE_API HandleAnimation {
     explicit HandleAnimation( const std::string& name = "" );
 
-    /// The handle's name.
+    /// The Handle's name.
     std::string m_name;
 
-    /// The list of KeyFramed transformed applied to the handle.
-    Core::Animation::KeyTransform m_anim;
+    /// The list of KeyFramed transforms applied to the Handle.
+    Core::Animation::KeyFramedValue<Transform> m_anim;
+
+    /// The AnimationTime for the Handle.
+    AnimationTime m_animationTime;
 };
 
-/// The AnimationData class stores all the HandleAnimations related to an object,
-/// e.g. the ones from all the skeleton bones.
+/**
+ * The AnimationData class stores all the HandleAnimation related to an
+ * animation of an object, one per animation Handle.
+ * \note Objects can have several AnimationData, one for each animation.
+ */
 class RA_CORE_API AnimationData : public AssetData
 {
   public:
     explicit AnimationData( const std::string& name = "" );
 
     ~AnimationData();
 
-    /// Returns the name of the object.
+    /**
+     * Sets the name of the animation.
+     */
     inline void setName( const std::string& name );
 
-    /// Return the AnimationTime linked to the object.
-    inline const Core::Animation::AnimationTime& getTime() const;
+    /// \name Time
+    /// \{
+
+    /**
+     * \returns the AnimationTime of the animation.
+     */
+    inline const AnimationTime& getTime() const;
+
+    /**
+     * Sets the AnimationTime of the animation.
+     */
+    inline void setTime( const AnimationTime& time );
 
-    /// Set the AnimationTime for the object.
-    inline void setTime( const Core::Animation::AnimationTime& time );
+    /**
+     * \returns the animation timestep.
+     */
+    inline AnimationTime::Time getTimeStep() const;
 
-    /// Return the animation timestep.
-    inline Core::Animation::Time getTimeStep() const;
+    /**
+     * Sets the animation timestep.
+    ￼ */
+    inline void setTimeStep( const AnimationTime::Time& delta );
+    /// \}
 
-    /// Set the animation timestep.
-    inline void setTimeStep( const Core::Animation::Time& delta );
+    /// \name Keyframes
+    /// \{
 
-    /// Return the number of transform.
+    /**
+     * \returns the number of HandleAnimations.
+     */
     inline uint getFramesSize() const;
 
-    /// Return the HandleAnimation, i.e. the whole animation frames.
+    /**
+     * \returns the list of HandleAnimations, i.e. the whole animation frames.
+     */
     inline std::vector<HandleAnimation> getFrames() const;
```

Comment:
Not sure the name is well chosen. HandleAnimation own a KeyFramedValues, defining all the animation frames wrt a handle. This method should be renamed getAllTheAnimatedParameter (or something less verbose as said previously by @dlyr)

Reply:
done